src/kubeobjects/query/dynakube.go

```diff
@@ -0,0 +1,83 @@
+package query
+
+import (
+	"context"
+
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/src/api/v1beta1"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	trustedCAKey = "certs"
+	proxyKey     = "proxy"
+	tlsCertKey   = "server.crt"
+)
+
+type DynakubeQuery struct {
+	clt       client.Client
+	namespace string
+}
+
+func NewDynakubeQuery(clt client.Client, namespace string) *DynakubeQuery {
+	return &DynakubeQuery{
+		clt:       clt,
+		namespace: namespace,
+	}
+}
+
+func (query *DynakubeQuery) Proxy(dynakube *dynatracev1beta1.DynaKube) (string, error) {
```

Comment:
nitpick: pointer receiver is not necessary

Reply:
nitpick2: why not pass in the context ? 

I know that our current use of context is "not optimal" but it would make it future proof, and more easily reusable in other parts of the code

Reply:
Added option to add context